lib/gooddata/lcm/actions/ensure_data_product.rb

```diff
@@ -0,0 +1,53 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class EnsureDataProduct < BaseAction
+      DESCRIPTION = 'Ensure DataProduct'
+
+      PARAMS = define_params(self) do
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'DataProduct to manage'
+        param :data_product, instance_of(Type::StringType), required: false
+      end
+
+      RESULT_HEADER = [
+        :data_product
+      ]
+
+      class << self
+        def call(params)
+          params = params.to_hash
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)